addons/hr_payroll_account/models/hr_payroll_account.py

```diff
@@ -0,0 +1,155 @@
+#-*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.tools import float_compare, float_is_zero
+
+
+class HrPayslipLine(models.Model):
+    _inherit = 'hr.payslip.line'
+
+    def _get_partner_id(self, credit_account):
+        """
+        Get partner_id of slip line to use in account_move_line
+        """
+        # use partner of salary rule or fallback on employee's address
+        partner_id = self.salary_rule_id.register_id.partner_id.id or self.slip_id.employee_id.address_home_id.id
```

Comment:
can you assign this in new variable 
partner = self.salary_rule_id.register_id.partner_id and then use this variable.